connectors/sources/google_drive.py

```diff
@@ -557,100 +871,182 @@ async def get_content(self, blob, timestamp=None, doit=None):
-            method="list",
-            corpora="allDrives",
-            q="trashed=false",
-            orderBy="modifiedTime desc",
-            fields="files,nextPageToken",
-            includeItemsFromAllDrives=True,
-            supportsAllDrives=True,
-            pageSize=100,
+
+        access_controls = []
+
+        async for permissions_page in self.google_drive_client.list_permissions(
+            file_id
         ):
-            yield file
+            permissions = permissions_page.get("permissions", [])
+            access_controls_page = self._process_permissions(permissions)
+            access_controls.extend(access_controls_page)
+
+        return access_controls
+
+    def _get_permissions_on_my_drive(self, file):
+        """Formats the access permissions on a my drive for the given object.
+
+        Args:
+            file (dict): The metadata of Google Drive file.
+
+        Returns:
+            list: A list of access permissions on my drive for a given file.
+        """
+
+        permissions = file.get("permissions", [])
+        access_controls = self._process_permissions(permissions)
+        return access_controls
 
-    def prepare_blob_document(self, blob, paths):
-        """Apply key mappings to the blob document.
+    def _process_permissions(self, permissions):
+        """Formats the access permission list for Google Drive object.
 
         Args:
-            blob (dict): Blob's metadata returned from the Drive.
+            permissions (list): List of permissions of Google Drive file returned from API.
 
         Returns:
-            dict: Blobs metadata mapped with the keys of `BLOB_ADAPTER`.
-        """
-
-        blob_document = {
-            "_id": blob.get("id"),
-            "created_at": blob.get("createdTime"),
-            "last_updated": blob.get("modifiedTime"),
-            "name": blob.get("name"),
-            "size": blob.get("size") or 0,  # handle folders and shortcuts
-            "_timestamp": blob.get("modifiedTime"),
-            "mime_type": blob.get("mimeType"),
-            "file_extension": blob.get("fileExtension"),
-            "url": blob.get("webViewLink"),
+            list: A list of processed access permissions for a given file.
+        """
+        processed_permissions = []
+
+        for permission in permissions:
+            permission_type = permission["type"]
+            access_permission = None
+
+            if _is_user_permission(permission_type):
+                access_permission = _prefix_user(permission.get("emailAddress"))
+            elif _is_group_permission(permission_type):
+                access_permission = _prefix_group(permission.get("emailAddress"))
+            elif _is_domain_permission(permission_type):
+                access_permission = _prefix_domain(permission.get("domain"))
+            elif _is_anyone_permission(permission_type):
+                access_permission = "anyone"
+            else:
+                self._logger.warning(
```

Comment:
Should `continue` here? otherwise `None` is appended to `processed_permissions`

Reply:
👍  In theory it shouldn't be possible as there is no other permission type according to the documentation but appending `None` permission can mess up the DLS logic so I will add the `continue` just to be safe.